File: matlab/lib/+IceSSL/ConnectionInfo.m
classdef ConnectionInfo < Ice.ConnectionInfo
    % ConnectionInfo   Summary of ConnectionInfo
    %
    % Provides access to the connection detaisl of an SSL connection.
    %
    % ConnectionInfo Properties:
    %   cipher - The negotiated cipher suite.
    %   certs - The certificate chain.
    %   verified - The certificate chain verification status.

    % Copyright (c) 2003-2018 ZeroC, Inc. All rights reserved.

    methods
        function obj = ConnectionInfo(underlying, incoming, adapterName, connectionId, cipher, ...
                                      certs, verified)
            if nargin == 0
                underlying = [];
                incoming = false;
                adapterName = '';
                connectionId = '';
                cipher = '';
                certs = [];
                verified = false;
            end
            obj = obj@Ice.ConnectionInfo(underlying, incoming, adapterName, connectionId);
            obj.cipher = cipher;
            obj.certs = certs;
            obj.verified = verified;
        end
    end
    properties(SetAccess=private)
        % cipher - The negotiated cipher suite.
        cipher char

        % certs - The certificate chain.
        certs

        % verified - The certificate chain verification status.
        verified logical
    end
end

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/SSL/ConnectionInfo.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice.SSL;

import java.security.cert.Certificate;

/** Provides access to the connection details of an SSL connection */
public final class ConnectionInfo extends com.zeroc.Ice.ConnectionInfo {
    /** The negotiated cipher suite. */
    public final String cipher;

    /** The certificate chain. */
    public final Certificate[] certs;

    /** The certificate chain verification status. */
    public final boolean verified;

    ConnectionInfo(com.zeroc.Ice.ConnectionInfo underlying, String cipher, Certificate[] certs, boolean verified) {
        super(underlying);
        this.cipher = cipher;
        this.certs = certs != null ? certs.clone() : null;
        this.verified = verified;
    }
}
